src/pages/index.js

```diff
@@ -32,71 +87,145 @@ const cardPopOut = new PopupWithImage({
+  newProfilePictureModal.open();
+});
+
+modalImageEditBtn.addEventListener("click", () => {
+  newImageModal.open();
 });
 
 const profileModal = new PopupWithForm({
   popupSelector: ".modal_type_profile",
   handleFormSubmit: (formData) => {
-    console.log(formData.name);
-    console.log(formData.job);
-    newUserInfo.setUserInfo({ name: formData.name, job: formData.job });
+    /*
+    profileModal.renderSaving(true);
+    */
+    api
+      .editUserInfo(formData)
+      .then((userData) => {
+        newUserInfo.setUserInfo({ name: userData.name, job: userData.about });
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+    /*
+      .finally(() => profileModal.renderSaving(false));
     profileModal.close();
+    */
   },
+  buttonText: "Save",
+});
+
+const newProfilePictureModal = new PopupWithForm({
+  popupSelector: ".modal_type_change-profile-picture",
+  handleFormSubmit: (formData) => {
+    /*
+    newProfilePictureModal.renderSaving(true);
+    */
+    api
+      .editUserProfilePicture(formData)
+      .then((userData) => {
+        newUserInfo.setUserPicture({ avatar: userData.avatar });
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+    /*
+      .finally(() => newProfilePictureModal.renderSaving(false));
+  
+    newProfilePictureModal.close();
+    */
+  },
+  buttonText: "Save",
 });
 
 const newImageModal = new PopupWithForm({
   popupSelector: ".modal_type_new-image",
   handleFormSubmit: (newData) => {
-    const addedCard = createCard(newData);
-    firstCards.addItem(addedCard);
+    /*
+    newImageModal.renderSaving(true);
+    */
+    api
+      .addNewCard(newData)
+      .then((cardData) => {
+        return createCard(cardData);
+      })
+      .then((addedCard) => cardArea.addItem(addedCard))
+      .catch((err) => {
+        console.error(err);
+      });
     modalImageForm.reset();
     newImageValidator.toggleButtonState();
+    /*
     newImageModal.close();
+    */
   },
+  buttonText: "Create",
 });
 
+profileModal.setEventListeners();
+newImageModal.setEventListeners();
+newProfilePictureModal.setEventListeners();
+
 function openModalImage(card) {
   cardPopOut.open(card.data.name, card.data.link);
 }
 
-modalProfileEditBtn.addEventListener("click", () => {
-  profileModal.open();
-  const data = newUserInfo.getUserInfo();
-  console.log(data);
-  modalName.value = data.name;
-  modalJob.value = data.job;
-  profileValidator.resetValidation();
-});
-
-modalImageEditBtn.addEventListener("click", () => {
-  /*
-  newImageValidator.resetValidation();
-  */
-  newImageModal.open();
-});
-
-profileModal.setEventListeners();
-newImageModal.setEventListeners();
-
 /* enable Validation */
 
 const profileValidator = new FormValidator(validationConfig, modalProfileForm);
 const newImageValidator = new FormValidator(validationConfig, modalImageForm);
+const newProfilePictureValidator = new FormValidator(
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object: `formValidators`.  Then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
// define an object for storing validators
const formValidators = {}

const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // Here you get the name of the form (if you don’t have it then you need to add it into each form in `index.html` first)
    const formName = formElement.getAttribute('name')

   // Here you store the validator using the `name` of the form
    formValidators[formName] = validator;
    validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```